code/modules/chemistry/tools/grenades.dm

```diff
@@ -170,6 +170,8 @@
 			if (src && !src.disposed)
 				a = get_area(src)
 				if(a.sanctuary) return
+				if(user && user.equipped() == src)
+					user.u_equip(src)
```

Comment:
Why not just call this irregardless

Reply:
`u_equip()` will call `dropped()` for non-human mobs regardless of if the item was equipped or not, which could lead to undesired behavior. For example: handspiders would be able to bypass their sprint restrictions if I removed this check. It should probably be done inside `u_equip()` instead, but I was reluctant to mess with it at first due to how many things could (potentially) break from it.